### ISSUE TITLE
XML Support?

### ISSUE THREAD

**Issue by therealssj** [infomaniac-amf/php#5]
Will this ever be supporting xml?
Apparently i have run into a situation where i need the xml support.

Or could i just get some info on how can i write the code myself to make it read xml format?

Thank you for the work! :)

**Comment by dannykopping** [comment]
Are you refering to the `XMLDocument` aspect of the spec?
http://wwwimages.adobe.com/www.adobe.com/content/dam/Adobe/en/devnet/amf/pdf/amf-file-format-spec.pdf

**Comment by therealssj** [comment]
umm no the XMLType , my data starts like 0a0b <-- this is the xml marker here right?
i looked at the spec.php and i saw it saw 0x0B // not supported.
Plus i get  "Class [object key name] could not be instatiated"  fatal error

**Comment by dannykopping** [comment]
Correct - on page 11, the `XML type` is defined with the marker of `0x0B`.
There are no current plans to support this, as it seems to me that the only benefit of having XML supported in the spec at all was because of ActionScript's `E4X` capabilities, which are not present in either PHP or JS. Thoughts?

**Comment by therealssj** [comment]
Well i am not really a very experienced person so do not really know what is E4X :P
Could i just get some info on how i could just start to maybe write the code myself?
And i guess given the fact i could find any library in php to read the data nobody really does use it.

**Comment by dannykopping** [comment]
There is really no reason to use the XML datatype at all in AMF, outside of using it in conjunction with ActionScript 3. You will save yourself a lot of trouble if you simply send through the data as a string, and use `SimpleXML` to parse the data.

**Comment by dannykopping** [comment]
More info on E4X
http://en.wikipedia.org/wiki/ECMAScript_for_XML

**Comment by therealssj** [comment]
I have no control over the data structure.
I am getting it from a game and i need to process it to extract the info.
And yes SimpleXml is way better for sure but amf files are smaller i guess that's why the amf format was used as the files are a bit on the big the side.

**Comment by dannykopping** [comment]
Right, well then you're probably best off looking at how any other datatype is processed by the library, and add your XML type, and then just process it as a string.

**Comment by therealssj** [comment]
May i ask some questions here if i get stuck in between?

**Comment by dannykopping** [comment]
Sure
